h2o-core/src/main/java/hex/grid/filter/FilterFunction.java

```diff
@@ -3,13 +3,26 @@
 
 import hex.Model;
 
+import java.util.function.Predicate;
+
 /**
  * Assumed to represent higher level of abstraction comparing to {@link java.util.function.Predicate}
- * 
- * By using {@link FunctionalInterface} it is possible to define lambda functions of a corresponding type. 
+ *
+ * By using {@link FunctionalInterface} it is possible to define lambda functions of a corresponding type.
  * Note: interface with {@link FunctionalInterface} will not provide full functional support ( e.g. no `compose` and `andThen` methods)
  */
 @FunctionalInterface
-public interface FilterFunction<MP extends Model.Parameters> {
-  Boolean apply(MP permutation);
+public interface FilterFunction<MP extends Model.Parameters> extends Predicate<MP>, Activatable<MP>, Resetable {
+
+  default boolean apply(MP permutation) {
+    return test(permutation);
+  }
```

Comment:
now that `activate` can make the decision internally, you don't need `apply` at all anymore.

Reply:
@seb-h2o why?

I need it to get global decision:
```
for (FilterFunction<MP> fun : _filterFunctions) {
      if (!fun.apply(permutation)) {
        evaluatePermutation = false;
        break;
      }
    }
    final boolean activateGlobalDecision = evaluatePermutation;
```

Reply:
@seb-h2o got it. After renaming it became clearer what you mean